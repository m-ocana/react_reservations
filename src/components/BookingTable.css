.table {
  text-align: left;
}

.table, th, td {
  border: 1px solid black;
  border-collapse: collapse;
}

th, td {
  padding: 4px;
}

.cancelled {
  text-decoration: line-through;
}
