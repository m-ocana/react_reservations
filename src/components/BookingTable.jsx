import React, { Component, PropTypes } from 'react'
import { formatName } from '../helpers/utils'
import style from './BookingTable.css'

export default class BookingTable extends Component {

  static PropTypes = {
    bookings: PropTypes.array,
    clickHandler: PropTypes.func,
    entryIndex: PropTypes.number
  }

  renderEntries = () => {
    const { bookings, entryIndex, clickHandler } = this.props
    return (
      bookings.map((booking, bookingIndex) => {
        const { time, partySize, seated, cancelled } = booking
        return (
          <tr
            onClick={(e) => clickHandler(e, entryIndex, bookingIndex)}
            className={cancelled ? style.cancelled : ''}
            key={bookingIndex}
          >
            <td>{formatName(booking)}</td>
            <td>{time}</td>
            <td>{partySize}</td>
            <td>{seated ? 'Y' : 'N'}</td>
          </tr>
        )
      })
    )
  }

  render() {
    return (
      <table className={style.table}>
        <thead>
          <tr>
            <th>Name</th>
            <th>Time</th>
            <th>Covers</th>
            <th>Seated</th>
          </tr>
        </thead>
        <tbody>
          { this.renderEntries() }
        </tbody>
      </table>
    )
  }
}
