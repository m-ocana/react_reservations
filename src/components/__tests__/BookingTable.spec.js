import renderComponent from '../../helpers/test-component'
import BookingTable from '../BookingTable'

const renderBookingTable = renderComponent(BookingTable)
const initialProps = {
  bookings: [
    {
      title: 'Mr',
      firstName: 'Fred',
      lastName: 'Bloggs',
      time: '16.00',
      partySize: 2,
      seated: true,
      cancelled: false,
      notes: ''
    },
    {
      title: 'Mr',
      firstName: 'John',
      lastName: 'Smith',
      time: '18.45',
      partySize: 1,
      seated: false,
      cancelled: true,
      notes: ''
    }
  ],
  entryIndex: 0,
  clickHandler: jest.fn()
}

describe('<BookingTable /> component', () => {
  describe('@renders', () => {
    it('should render BookingTable', () => {
      expect(renderBookingTable(initialProps).getTree())
        .toMatchSnapshot()
    })
  })

  describe('@events', () => {
    describe('on table row click', () => {
      it('should trigger onClickHander', () => {
        const { wrapper, instance } = renderBookingTable(initialProps)
        wrapper.find('tbody tr').first().simulate('click')
        expect(instance.props.clickHandler).toHaveBeenCalled()
      })
    })
  })
})
