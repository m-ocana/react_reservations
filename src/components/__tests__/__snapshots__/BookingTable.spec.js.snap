// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingTable /> component @renders should render BookingTable 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Time
      </th>
      <th>
        Covers
      </th>
      <th>
        Seated
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className=""
      onClick={[Function]}
    >
      <td>
        Mr Fred Bloggs
      </td>
      <td>
        16.00
      </td>
      <td>
        2
      </td>
      <td>
        Y
      </td>
    </tr>
    <tr
      className="cancelled"
      onClick={[Function]}
    >
      <td>
        Mr John Smith
      </td>
      <td>
        18.45
      </td>
      <td>
        1
      </td>
      <td>
        N
      </td>
    </tr>
  </tbody>
</table>
`;
